src/org/zaproxy/zap/extension/scripts/ExtensionScriptsUI.java

```diff
@@ -659,11 +663,26 @@ public Writer getOutputWriter() {
 	}
 	
 	@Override
-	public void engineAdded(ScriptEngineWrapper scriptEngineWrapper) {
+	public void engineAdded(final ScriptEngineWrapper scriptEngineWrapper) {
 		if (getView() == null) {
 			return;
 		}
 
+		if (!EventQueue.isDispatchThread()) {

```

Comment:
Am i missing something? Commit message says do it in EDT but this condition is not EDT?

Reply:
Right, if not EDT do/run in EDT. Otherwise it's already EDT and can continue as usual.

Reply:
So EventQueue.invokeAndWait ends up with it in EDT? 

(Sorry I'm sure I've asked about this before but the logic/behaviour isn't intuitive.)

Reply:
Yeah, that call executes the runnable in the EDT.